### ISSUE TITLE
Change URL once primary and/or secondary colors have been selected

### ISSUE THREAD

**Issue by edelstone** [edelstone/material-palette-generator#1]
Adding the colors to a specialized URL would allow the link to be shared and bookmarked easily.

Got this via email:

> Thanks for the handy Material Design Palette Generator. It would be great if we could bookmark it with the primary/secondary colors in the URL, like https://materialpalettes.com/?primary=#2f5597. Is that something you would consider implementing?

**Comment by vkottler** [comment]
Hmm, honestly what would be awesome would be some kind of hosted API that accepts "hex color(s)" (maybe "primary", "secondary") and sends back some JSON data with the palette(s) in RGB-hex. I'm working on an offline generator for this, but I wish Google had the source for this in one of their [material components](https://github.com/material-components/material-components-web/tree/master/packages). The "theme" component is not very sophisticated and requires manual entry for the actual colors.

Got as far as looking through ["Color Tool"](https://material.io/resources/color/#!/?view.left=0&view.right=1&primary.color=60ff48) source to find that in the end they're literally just calling [chroma.js](https://github.com/gka/chroma.js) `lighten` and `darken` (since _that_ tool doesn't do the palette generator...).

Anyway, stumbled upon this project on this quest. Thanks for hosting it!

**Comment by JorensM** [comment]
So I tinkered around with the code, and I'm not sure what's going on exactly, but it looks like the function `U()` in `main.js`, which supposedly takes in a hsl value, doesn't seem to be accurate. in the place where we set the primary color, if I replace the `U` function's arguments with `0`, `1` and `0.5`, this is the color I get:

![same-hsl-on-app](https://github.com/edelstone/material-palette-generator/assets/17122123/5cfa18be-158e-40df-b630-33b355af52d2)

While actually the color with HSL values 0, 1 and 0.5 looks like this:

![actual-red-hsl](https://github.com/edelstone/material-palette-generator/assets/17122123/6123bfd8-70d6-4eb0-a3d4-f395291bdbf4)

So yeah, I'm not sure why that's happening.

**Comment by edelstone** [comment]
@JorensM Yeah me either. You may have read in the README that it's all Google's obfuscated code doing the color calculations. Likely not something we should attempt to change but feel free to work around it.

**Comment by JorensM** [comment]
@edelstone I guess I'll pass this one since I don't want to break something somewhere else haha. Maybe someone braver than me will come along and give it a go!

**Comment by edelstone** [comment]
@JorensM not a problem. Thanks for looking!
